test/tools/parser_throughput/main.cc

```diff
@@ -149,21 +175,32 @@ int main(int argc, char** argv)
       }
     }
   }
-  else
+  else if (type == parser_type::dsjson)
   {
     DecisionServiceInteraction interaction;
-    for (const auto& line : lines)
+    for (const auto& line : file_contents_as_lines)
     {
       VW::multi_ex examples;
       examples.push_back(&VW::get_unused_example(vw));
       VW::read_line_decision_service_json<false>(*vw, examples, const_cast<char*>(line.data()), line.length(), false,
           (VW::example_factory_t)&VW::get_unused_example, (void*)vw, &interaction);
-      VW::multi_ex result;
-      result.reserve(examples.size());
-      for (size_t i = 0; i < examples.size(); ++i) { result.push_back(examples[i]); }
-      // TODO - finish_example should support a v_array as input.
-      VW::finish_example(*vw, result);
+      VW::finish_example(*vw, examples);
+    }
+  }
+  else
+  {
+#ifdef VW_BUILD_CSV
+    VW::multi_ex examples;
+    examples.push_back(&VW::get_unused_example(vw));
+    while (VW::parsers::parse_csv_examples(vw, file_contents_as_io_buf, examples) != 0)
+    {
+      VW::finish_example(*vw, *examples[0]);
+      examples.clear();
+      examples.push_back(&VW::get_unused_example(vw));
     }
+#else
```

Comment:
To fix the `Assertion failed: m_pool.size() == size()` error.

```suggestion
    VW::finish_example(*vw, *examples[0]);
#else
```

```log
./parser_throughput --type csv --data data.csv
17994212 bytes parsed in 829498μs
21.6929MB/s
Assertion failed: m_pool.size() == size(), file D:\RLOSFest\vowpal_wabbit\vowpalwabbit\core\include\vw\core\object_pool.h, line 52
```